available_brick_points/app.py

```diff
@@ -0,0 +1,98 @@
+__author__ = "Anand Krishnan Prakash"
+__email__ = "akprakash@lbl.gov"
+
+import pymortar
+import datetime
+import pandas as pd
+import argparse
+
+def get_query():
+    query = """SELECT ?point ?point_type WHERE { ?point rdf:type/rdfs:subClassOf* brick:Point . ?point rdf:type ?point_type . };"""
+    return query
+
+def get_all_points(client, site=None):
+    query = get_query()
+
+    resp = client.qualify([query])
+    if resp.error != "" :
+        print("ERROR: ", resp.error)
+
+    if site!=None and site not in resp.sites:
+        return pd.DataFrame()
+
+    if len(resp.sites) == 0:
+        return pd.DataFrame()
+
+    if site !=None:
+        sites = [site]
+    else:
+        sites = resp.sites
+
+    points_view = pymortar.View(
+        sites=sites,
+        name="point_type_data",
+        definition=query,
+    )
+
+    time_params = pymortar.TimeParams(
+        start="2017-01-01T00:00:00Z",
+        end="2018-01-01T00:00:00Z",
+    )
+
+    request = pymortar.FetchRequest(
+        sites=sites,
+        views=[points_view],
+        time=time_params
+    )
+
+    response = client.fetch(request)
+    if len(response.tables) == 0:
+    	return pd.DataFrame()
+
+    sites = [site[0] for site in response.query('select distinct site from point_type_data')]
+
+    df_list = []
+    for site in sites:
+        q = """
+                SELECT point, point_type
+                FROM point_type_data
+                WHERE site = "{0}";
+            """.format(site)
+        out = response.query(q)
+        points = []
+        point_types = []
+        for point, point_type in out:
```

Comment:
I recognize this is some common code to write and I have a plan to make this easier (https://todo.sr.ht/~gabe/mortar/27)

Reply:
Yes, I believe that it'll be nice to have this as a standard function or similar in pymortar rather than an app. If you'll be adding that, I'll close this pull request. Does that work?

Reply:
There's some new functionality in pymortar-1.0.3 documented [here](https://mortardata.org/pymortar/pymortar.html#module-pymortar.result) that might simplify the logic of parsing the Brick metadata (specifically take a look at [`Result.view`](https://mortardata.org/pymortar/pymortar.html#pymortar.result.Result.view)

Reply:
The functionality I'm talking about is used in your app, but doesn't supplant what its doing, so you don't need to close this PR